src/Avalonia.Controls.TreeDataGrid/Selection/TreeDataGridRowSelectionModel.cs

```diff
@@ -110,6 +113,99 @@ void ITreeDataGridSelectionInteraction.OnKeyDown(TreeDataGrid sender, KeyEventAr
 
         }
 
+        protected void HandleCharKey(string? text, TreeDataGrid treeDataGrid, int selectedRowIndex)
+        {
+            if (text != null && treeDataGrid.Columns != null)
+            {
+                var typedChar = text[0];
+
+                long now = DateTime.Now.Ticks;
+                long time = 0;
+                if (_lastCharPressedTime > 0)
+                {
+                    time = now - _lastCharPressedTime;
+                }
+
+                string candidatePattern;
+                if (time < 5000000)
+                {
+                    if (_typedWord.Length == 1 && typedChar == _typedWord[0])
+                    {
+                        candidatePattern = _typedWord;
+                    }
+                    else
+                    {
+                        candidatePattern = _typedWord + typedChar;
+                    }
+                }
+                else
+                {
+                    candidatePattern = typedChar.ToString();
+                }
+                foreach (var column in treeDataGrid.Columns)
+                {
+                    if (column is ITextSearchable textSearchableColumn && textSearchableColumn.IsTextSearchEnabled == true)
+                    {
+                        Search(treeDataGrid, candidatePattern, selectedRowIndex, textSearchableColumn);
+                    }
+                    else if (column is HierarchicalExpanderColumn<TModel> hierarchicalColumn &&
+                        hierarchicalColumn.Inner is ITextSearchable textSearchableColumn2)
+                    {
+                        Search(treeDataGrid, candidatePattern, selectedRowIndex, textSearchableColumn2);
+                    }
+
+                }
+                _lastCharPressedTime = now;
+            }
+        }
+
+        private void Search(TreeDataGrid treeDataGrid, string candidatePattern, int selectedRowIndex, ITextSearchable column)
+        {
+            var textColumn = column as ISearchableTextColumn<TModel>;
+            var templateColumn = column as ISearchableTemplateColumn<TModel>;
+            var found = false;
+            var selector = textColumn == null ? templateColumn!.TextSearchValueSelector : textColumn.ResolveStingValueFromModel;
```

Comment:
If `selector` is null here I think we can exit early as the `SearchAndSelectRow` method does nothing in this case.